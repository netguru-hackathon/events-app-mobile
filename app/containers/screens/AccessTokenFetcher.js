import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  View,
  StyleSheet,
  Text,
} from 'react-native';
import I18n from '../../utils/translations';

// actions
import { authUser } from '../../actions/auth';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

class AccessTokenFetcher extends Component {
  static propTypes = {
    authUser: PropTypes.func.isRequired,
    navigation: PropTypes.shape({
      state: PropTypes.object.isRequired,
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  componentWillMount() {
    const { navigation } = this.props;
    const { code } = navigation.state.params;

    this.props.authUser(code)
      .then(navigation.navigate('DashboardNavigator'))
      .catch((error) => {
        // eslint-disable-next-line
        console.log(error);
      });
  }

  render() {
    return (
      <View
        style={styles.container}
      >
        <Text>
          {I18n.t('loading')}
        </Text>
      </View>
    );
  }
}

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  authUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(AccessTokenFetcher);
