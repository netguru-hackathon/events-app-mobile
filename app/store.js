import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import axiosMiddleware from 'redux-axios-middleware';

import axiosClient from './api/client';
import Reducers from './reducers/index';

const store = applyMiddleware(
  thunk,
  axiosMiddleware(axiosClient, { returnRejectedPromiseOnError: true }),
)(createStore)(Reducers);

export default store;
