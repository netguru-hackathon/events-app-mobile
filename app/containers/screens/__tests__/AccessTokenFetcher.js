import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AccessTokenFetcher from '../AccessTokenFetcher';
import store from '../../../store';

test('renders correctly', () => {
  const tree = renderer.create(
    <AccessTokenFetcher
      store={store}
      navigation={{
        state: {
          params: {
            code: 'code',
          },
        },
      }}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
