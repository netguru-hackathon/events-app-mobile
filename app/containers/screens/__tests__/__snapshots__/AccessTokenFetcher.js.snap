// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "alignItems": "center",
        "height": 80,
        "justifyContent": "center",
        "padding": 8,
      }
    }
  />
</View>
`;
